src/core/Akka.Streams/Implementation/Fusing/Ops.cs

```diff
@@ -2577,19 +2564,34 @@ public override void OnPush()
                 try
                 {
                     var task = _stage._mapFunc(message);
-                    var holder = new Holder<TOut>(message, NotYetThere, _taskCallback);
+                    Debug.Assert(message != null, nameof(message) + " != null");
+                    var holder = new Holder<TOut>(message!, NotYetThere);
                     _buffer.Enqueue(holder);
 
                     // We dispatch the task if it's ready to optimize away
                     // scheduling it to an execution context
                     if (task.IsCompleted)
-                    {
-                        holder.SetElement(Result.FromTask(task));
-                        HolderCompleted(holder);
+                    {;
```

Comment:
```suggestion
                    {
```

Looks like there is an extra `;`

Reply:
Good catch - fixed